www/profile/inner.js

```diff
@@ -101,40 +101,57 @@ define([
     var sFrameChan;
 
     var addViewButton = function ($container) {
-        if (APP.readOnly) {
-            return;
-        }
-
         var hash = common.getMetadataMgr().getPrivateData().hashes.viewHash;
         var url = APP.origin + '/profile/#' + hash;
-
-        $('<button>', {
-            'class': 'btn '+VIEW_PROFILE_BUTTON,
-        }).text(Messages.profile_viewMyProfile).click(function () {
-            window.open(url, '_blank');
-        }).appendTo($container);
-
-        $('<button>', {
-            'class': 'btn btn-primary '+VIEW_PROFILE_BUTTON,
-        }).append(h('i.fa.fa-shhare-alt'))
-          .append(h('span', Messages.shareButton))
-          .click(function () {
-            Clipboard.copy(url, (err) => {
-                if (!err) { UI.log(Messages.shareSuccess); }
+        // Only add view and share buttons if not read-only
+        if (!APP.readOnly) {
+            var $blockView = $('<div>', {class: LINK_ID}).appendTo($container);
+            var $button = $('<button>', {
+                'class': 'btn ' + VIEW_PROFILE_BUTTON,
+                'aria-labelledby': 'cp-view-button'
+            }).appendTo($blockView);
+            $button.append(
+                h('span', { id: 'cp-view-button' }, Messages.profile_viewMyProfile)
+            );
+            $button.click(function () {
+                window.open(url, '_blank');
             });
-        }).appendTo($container);
+        }
+        
+        // Always create the description container
+        var $blockDescription = $('<div>', {id: DESCRIPTION_ID}).appendTo($container);
+        APP.$descriptionContainer = $blockDescription;
+        APP.$description = $('<div>', {
```

Comment:
Please try to use hyperscript when you add or rewrite code and you don't need to jQuery tools. It's faster to render, especially on mobile.